pandas/plotting/_matplotlib/core.py

```diff
@@ -124,13 +127,42 @@ def __init__(
         table=False,
         layout=None,
         include_bool=False,
+        column: IndexLabel | None = None,
         **kwds,
     ):
 
         import matplotlib.pyplot as plt
 
         self.data = data
-        self.by = by
+
+        # if users assign an empty list or tuple, treat them as None
+        # then no group-by will be conducted.
```

Comment:
why is by allowed to be an empty list/tuple?

Reply:
you are right. I initially thought that is the behaviour of current `df.hist` and `df.box`, but I am wrong. I change to raise an error exactly the same as current `df.hist` and `df.box`.

I also changed the inline comment above this line and align with the change, and also add tests to reflect the changes.